.buildkite/scripts/common/run-main-e2e-tests.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# On any command failures, make sure that diagnostics are copied from bucket to local Buildkite environment
+# to be uploaded as artifacts.
+onError() {
+    # On error, copy any artifacts from the test run locally to be picked up as buildkite artifacts
+    if gsutil ls "gs://eck-e2e-buildkite-artifacts/jobs/$BUILDKITE_PIPELINE_NAME/$BUILDKITE_BUILD_NUMBER/eck-diagnostic*.zip" ; then
+        gsutil cp "gs://eck-e2e-buildkite-artifacts/jobs/$BUILDKITE_PIPELINE_NAME/$BUILDKITE_BUILD_NUMBER/eck-diagnostic*.zip" .
```

Comment:
Do we need to remove the artefacts after copying or to avoid filling up this bucket over time?